### PR TITLE
implementing COF logic

### PR DESCRIPTION
**Pull request by jlieberm**
This PR implement the Constrained Optimal Filter method for energy estimation under a flag --estimationMethod COF

### CODE REVIEW

File: reconstruction/CaloCellBuilder/src/ConstrainedOptimalFilter.cxx [hunk 1]
```diff
@@ -36,21 +63,86 @@ StatusCode ConstrainedOptimalFilter::finalize()
...
+    if (passedSamples[i]){
+      for (int j = 0; j < m_nsamples; j++) {
+        G[k][j]= saveH[i][j];
+        saveG[k][j] = saveH[i][j];
+      }
+      k++;
+    }
+  }
+  auto a_hat_hat = ((saveG*G.T()).Invert())*saveG*tpulse;
+  // float energy=0.0;
```
Comment (micaelverissimo) [inline reconstruction/CaloCellBuilder/src/ConstrainedOptimalFilter.cxx]: @jlieberm if we don't need of these commented lines, could you remove them?

**Comment by micaelverissimo** [comment]
I think the modifications are clean, just take a look at the commented lines that I comment.

**Comment by jlieberm** [comment]
Yes, indeed. Just removed the commented lines.

**Review by micaelverissimo — APPROVED** [review]
It's fine for me.

**Pull request merged**